### PR TITLE
MNT Use React Testing Library

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1506

### CODE REVIEW

File: client/src/components/TOTP/tests/Register-test.js [hunk 1]
```diff
@@ -1,193 +1,118 @@
...
+    <Register {...makeProps({
+      error: 'Something went wrong'
+    })}
+    />
+  );
+  const back = await screen.findByText('Back');
+  fireEvent.click(back);
+  const el = await screen.findByText('How to use authenticator apps.');
+  expect(el).not.toBeNull();
+});
```
Comment (ChloeSartorelli) [inline client/src/components/TOTP/tests/Register-test.js]: This isn't actually checking if an error has been cleared - we should validate that an element matching `.mfa-totp__errors` isn't there.

File: client/src/components/TOTP/tests/Register-test.js
Comment (ChloeSartorelli) [inline client/src/components/TOTP/tests/Register-test.js:1]: Missing "goes back to the previous screen from the initial screen when clicking 'Back'" test

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: client/src/components/TOTP/tests/Register-test.js
Comment (emteknetnz) [reply client/src/components/TOTP/tests/Register-test.js:1]: Covered by the 'Register handleBack() calls the onBack prop' test

File: client/src/components/TOTP/tests/Register-test.js (hunk 1 above)
Comment (emteknetnz) [reply client/src/components/TOTP/tests/Register-test.js]: Updated

**Review by GuySartorelli — APPROVED** [review]

**Pull request merged**